stablehlo/dialect/TypeInference.cpp

```diff
@@ -3802,15 +3759,14 @@ LogicalResult verifyScatterOp(std::optional<Location> location,
   bool scatterIndicesTypeRanked = scatterIndicesType.isa<RankedTensorType>();
 
   // P1.
-  if (scatterIndicesTypeRanked) {
+  if (scatterIndicesTypeRanked)
     if (indexVectorDim > scatterIndicesType.getRank() || indexVectorDim < 0)
       return emitOptionalError(
           location,
           "expects scatter index leaf dimension to be within [0, "
           "rank(scatter_indices) + 1. rank(scatter_indices) is ",
           scatterIndicesType.getRank(), " and scatter index leaf dimension is ",
           indexVectorDim, ".");
```

Comment:
```
// Use braces on the outer `if` to avoid a potential dangling `else`
// situation.
if (isa<VarDecl>(D)) {
  if (shouldProcessAttr(A))
    handleAttr(A);
}
```